src/base/RegexPattern.cc

```diff
@@ -8,34 +8,33 @@
 
 #include "squid.h"
 #include "base/RegexPattern.h"
-#include "base/TextException.h"
 #include "debug/Stream.h"
 #include "sbuf/Stream.h"
 
 #include <iostream>
-#include <utility>
 
-RegexPattern::RegexPattern(const SBuf &aPattern, const int aFlags):
-    pattern(aPattern),
-    flags(aFlags)
+RegexPattern::RegexPattern(const SBuf &aPattern, const std::regex::flag_type aFlags):
+    pattern(aPattern)
 {
-    memset(&regex, 0, sizeof(regex)); // paranoid; POSIX does not require this
-    if (const auto errCode = regcomp(&regex, pattern.c_str(), flags)) {
-        char errBuf[256];
-        // for simplicity, ignore any error message truncation
-        (void)regerror(errCode, &regex, errBuf, sizeof(errBuf));
-        // POSIX examples show no regfree(&regex) after a regcomp() error;
-        // presumably, regcom() frees any allocated memory on failures
-        throw TextException(ToSBuf("POSIX regcomp(3) failure: (", errCode, ") ", errBuf,
-                                   Debug::Extra, "regular expression: ", pattern), Here());
+    try {
+        regex = std::regex(pattern.rawContent(), pattern.length(), aFlags|std::regex::nosubs|std::regex::optimize);
+    } catch (const std::regex_error &e) {
+        throw TextException(ToSBuf(e.what(), ": ", pattern), Here());
```

Comment:
Please catch and annotate _all_ exceptions (and not just std::regex_error) and supply local context instead of relying on the caught exception (that we do not control) to be descriptive enough to explain what configuration step/action has failed. The official code got this right (where applicable), and we should continue to use that error reporting approach. For example (untested, meant as a sketch; as always where I do not indicate otherwise):

```suggestion
    } catch (...) {
        throw TextException(ToSBuf("regex compilation failure:", CurrentException,
                                   Debug::Extra, ": regular expression: ", pattern), Here());
```

Reply:
This change request will also make some errors more readable (because the code will assume much less about the STL error text that we do not control). For example, this particular STL implementation ends errors with periods (`.`), resulting in slightly odd/misleading `escape character.: ...` output:

```
2023/02/20 14:10:04| ERROR: configuration failure: Unexpected escape character.: "foo\"
    exception location: src/base/RegexPattern.cc(22) RegexPattern
```

BTW, please disclose regex pattern (enforcement) changes in the PR description. See our CI tests (from where the above error was pasted).

Reply:
Done in [0afa2f5](https://github.com/squid-cache/squid/pull/1232/commits/0afa2f57534d6cc9dca9a96b395a133b1904df78)

Reply:
This change request was not fully addressed by commit [0afa2f5](https://github.com/squid-cache/squid/pull/1232/commits/0afa2f57534d6cc9dca9a96b395a133b1904df78), but since it is now outdated, I will open another/fresh change request about the same problem.